### PR TITLE
Metrics and Metadata Service Export when needed

### PR DESCRIPTION
**Pull request by AlbumenJ**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by AlbumenJ** [comment]
@songxiaosheng PTAL

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java [hunk 1]
```diff
@@ -239,6 +239,10 @@ public interface CommonConstants {
 
     String REMOTE_METADATA_STORAGE_TYPE = "remote";
```
Comment (songxiaosheng) [inline dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java:241]: 元数据目前一共有两种注册方式，集中式注册到元数据中心的remote配置，和默认分布式RPC配置，这个新增的类型是干啥的没看懂

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java [hunk 2]
```diff
@@ -748,11 +748,15 @@ public void prepareInternalModule() {
     }
 
     private void exportMetricsService() {
-        try {
-            metricsServiceExporter.export();
-        } catch (Exception e) {
-            logger.error(LoggerCodeConstants.COMMON_METRICS_COLLECTOR_EXCEPTION, "", "",
-                "exportMetricsService an exception occurred when handle starting event", e);
+        boolean exportMetrics = applicationModel.getApplicationConfigManager().getMetrics()
+            .map(MetricsConfig::getExportMetricsService).orElse(true);
```
Comment (songxiaosheng) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java:752]: 这里看着没啥问题

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java [hunk 3]
```diff
@@ -239,6 +239,10 @@ public interface CommonConstants {
...
 
+    String INTERFACE_REGISTER_MODE = "interface";
```
Comment (songxiaosheng) [inline dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java:242]: +

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java (hunk 1 above)
Comment (songxiaosheng) [reply dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java:241]: String INTERFACE_REGISTER_MODE = "interface"; 这个

Comment (AlbumenJ) [reply dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java:241]: 这个是是否开启用应用级服务发现的
在 dubbo.application.register-mode 里面

**Pull request merged**